android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt

```diff
@@ -167,6 +167,7 @@ class OkHttpTest {
       }
     } finally {
       Security.removeProvider("Conscrypt")
+      client.close()
```

Comment:
FWIW these tests triggered the overall bug.  They have a client build unchained from OkHttpClientTestRule.